### PR TITLE
Add Toggle to Allow Activation Commands to Use Different Namespaces

### PR DESCRIPTION
**Pull request by dubeejw**
- Add a toggle to use the namespace passed for activation commands
- Refactor qualified name parsing
- Remove dead code
- Fixes https://github.com/openwhisk/openwhisk/issues/1344

### CODE REVIEW

**Comment by rabbah** [comment]
See my comment in #1344. We can't merge this without introducing another "bug". This is blocked by the work on disambiguating activation namespaces (epic #1107).

**Comment by dubeejw** [comment]
I'll remove the changes that prevent the namespace from defaulting to "_" for activations. I would still like to get this PR merged as it contains changes to remove a lot of dead code.

**Comment by rabbah** [comment]
> I'll remove the changes that prevent the namespace from defaulting to "_" for activations

how about: guard it with a feature toggle? we will need this, sooooon.

**Comment by dubeejw** [comment]
PG1 572.

**Comment by dubeejw** [comment]
PG2 393.

File: tools/cli/go-whisk-cli/commands/util.go [hunk 1]
```diff
@@ -68,33 +68,41 @@ Examples:
       /ns/foo => qName {namespace: ns, entityName: foo}
       /ns/pkg/foo => qName {namespace: ns, entityName: pkg/foo}
 */
-func parseQualifiedName(name string) (qName qualifiedName, err error) {
+func parseQualifiedName(name string) (QualifiedName) {
```
Comment (mdeuser) [inline tools/cli/go-whisk-cli/commands/util.go]: No longer returning an error... Isn't it possible for an entity name to be incorrectly formatted?  For example:
  `wsk action get ""`
The deprecated python CLI returned `error: entity name missing, did you mean to list collection`

File: tools/cli/go-whisk/whisk/activation.go [hunk 2]
```diff
@@ -70,9 +70,13 @@ type Log struct {
     Time   string `json:"time,omitempty"`
 }
 
+const USE_DEFAULT_NAMESPACE = true
```
Comment (mdeuser) [inline tools/cli/go-whisk/whisk/activation.go]: Would this ever be `false`?

File: tools/cli/go-whisk-cli/commands/package.go [hunk 3]
```diff
@@ -413,31 +357,11 @@ var packageListCmd = &cobra.Command{
...
-      if len(ns) == 0 {
-        whisk.Debug(whisk.DbgError, "An empty namespace in the package name '%s' is invalid \n", args[0])
-        errStr := fmt.Sprintf(
-          wski18n.T("No valid namespace detected. Run 'wsk property set --namespace' or ensure the name argument is preceded by a \"/\""))
-        werr := whisk.MakeWskError(errors.New(errStr), whisk.EXITCODE_ERR_GENERAL, whisk.DISPLAY_MSG, whisk.DISPLAY_USAGE)
-        return werr
-      }
-
-      client.Namespace = ns
+      qualifiedName := parseQualifiedName(args[0])
```
Comment (mdeuser) [inline tools/cli/go-whisk-cli/commands/package.go]: Technically, the following command should return an error as the argument is not a namespace
`package list /mdeuser_dev/kjous/lajfdlj/aljdf`

File: tools/cli/go-whisk-cli/commands/trigger.go [hunk 4]
```diff
@@ -548,28 +441,9 @@ var triggerListCmd = &cobra.Command{
...
-            if len(ns) == 0 {
-                whisk.Debug(whisk.DbgError, "Namespace is missing from '%s'\n", args[0])
-                errStr := fmt.Sprintf(
-                    wski18n.T("No valid namespace detected. Run 'wsk property set --namespace' or ensure the name argument is preceded by a \"/\""))
-                werr := whisk.MakeWskError(errors.New(errStr), whisk.EXITCODE_ERR_GENERAL, whisk.DISPLAY_MSG, whisk.DISPLAY_USAGE)
-                return werr
-            }
-            client.Namespace = ns
-            whisk.Debug(whisk.DbgInfo, "Using namespace '%s' from argument '%s''\n", ns, args[0])
+            qualifiedName := parseQualifiedName(args[0])
```
Comment (mdeuser) [inline tools/cli/go-whisk-cli/commands/trigger.go]: See above comment re: invalid namespace argument

File: tools/cli/go-whisk-cli/commands/util.go (hunk 1 above)
Comment (dubeejw) [inline tools/cli/go-whisk-cli/commands/util.go]:
```
$ wsk action get ""
error: Invalid argument(s). An action name is required.
Run 'wsk --help' for usage.
```

File: tools/cli/go-whisk/whisk/activation.go (hunk 2 above)
Comment (dubeejw) [inline tools/cli/go-whisk/whisk/activation.go]: @mdeuser, this is temporary until we can use non-default namespaces with the activation commands.

**Pull request closed without merging**